### PR TITLE
Refactor bottomNavigationBar parameter of HomePage.

### PR DESCRIPTION
**Pull request by georgecabaccang**
- Extract NavigationBar widget to its own class called BottomNavigation.
- Create BottomNavDestination class where the NavgationDestination list is.
- Use BottomNavDestinations as value for destinations parameter of BottomNavigation in HomePage.

### CODE REVIEW

File: lib/pages/home/utils/bottom_nav/bottom_navigation.dart [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import 'package:flutter/material.dart';
+import 'package:school_erp/pages/home/utils/bottom_nav/bottom_nav_destinations.dart';
+
+typedef VoidCallbackWithArg = void Function(int);
```
Comment (Kelv1nG) [inline lib/pages/home/utils/bottom_nav/bottom_navigation.dart:4]: its ok to use inline here as long as the type def is simple enough

https://dart.dev/effective-dart/design#prefer-inline-function-types-over-typedefs

**Review by Kelv1nG — APPROVED** [review]

File: lib/pages/home/utils/bottom_nav/bottom_navigation.dart (hunk 1 above)
Comment (georgecabaccang) [reply lib/pages/home/utils/bottom_nav/bottom_navigation.dart]: Resolved.

File: lib/pages/home/app_body/views/settings/widgets/helpers/settings_list.dart [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+          title: Text('School Profile'),
+          onTap: () {},
+        ),
+        ListTile(
+          leading: Icon(Icons.lock),
+          title: Text('Change Password'),
+          onTap: () {
+            Navigator.push(
+              context,
+              createSlideRoute(ChangePasswordPage()),
```
Comment (Kelv1nG) [inline lib/pages/home/app_body/views/settings/widgets/helpers/settings_list.dart:34]: convert to stateless widget

File: lib/pages/home/app_body/views/settings/widgets/helpers/settings_list.dart [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        ),
+        ListTile(
+          leading: Icon(Icons.logout_outlined),
+          title: Text('Logout'),
+          onTap: () => LogoutModal(context).show(),
+        ),
+      ],
+    );
+  }
+}
```
Comment (Kelv1nG) [inline lib/pages/home/app_body/views/settings/widgets/helpers/settings_list.dart:83]: convert to stateless widget

File: lib/pages/home/app_body/views/settings/widgets/helpers/settings_list.dart [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            Navigator.push(
+              context,
+              createSlideRoute(ChangePasswordPage()),
+            );
+          },
+        ),
+      ],
+    );
+  }
+}
```
Comment (Kelv1nG) [inline lib/pages/home/app_body/views/settings/widgets/helpers/settings_list.dart:41]: convert to stateless widget

**Review by Kelv1nG — CHANGES_REQUESTED** [review]
requesting changes

File: lib/pages/home/app_body/views/settings/widgets/helpers/settings_list.dart (hunk 4 above)
Comment (georgecabaccang) [reply lib/pages/home/app_body/views/settings/widgets/helpers/settings_list.dart]: Resolved. 
Fix is here: ef785e57de9168a529674bc33d4557157f5b5bfe
File: 
- **lib/pages/home/app_body/views/settings/widgets/helpers/settings_list.dart**

File: lib/pages/home/app_body/views/settings/widgets/helpers/settings_list.dart (hunk 3 above)
Comment (georgecabaccang) [reply lib/pages/home/app_body/views/settings/widgets/helpers/settings_list.dart]: Resolved. 
Fix is here: ef785e57de9168a529674bc33d4557157f5b5bfe
File: 
- **lib/pages/home/app_body/views/settings/widgets/helpers/settings_list.dart**

**Pull request merged**